File: FE/WEB/app/models/user-preference.js
import Model, { attr } from '@ember-data/model';

export default class UserPreferenceModel extends Model {
  @attr('string', { defaultValue: '' }) userId;
  @attr('number', { defaultValue: 0 }) tenantId;
  @attr('string', { defaultValue: '' }) prefKey;
  @attr('string', { defaultValue: '' }) prefValue;
  @attr('date-utc', {
    defaultValue() {
      return new Date();
    },
  })
  createdAt;
  @attr('date-utc', {
    defaultValue() {
      return new Date();
    },
  })
  updatedAt;
}

File: BE/Src/MIT.Fwk.Infrastructure/Entities/UserPreference.cs
using JsonApiDotNetCore.Resources;
using JsonApiDotNetCore.Resources.Annotations;
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace MIT.Fwk.Infrastructure.Entities
{
    [Table("UserPreference")]
    public class UserPreference : Identifiable<string>
    {
        [Attr] public string UserId { get; set; }
        [Attr] public int TenantId { get; set; }
        [Attr] public string PrefKey { get; set; }
        [Attr] public string PrefValue { get; set; }
        [Attr] public DateTime? CreatedAt { get; set; }
        [Attr] public DateTime? UpdatedAt { get; set; }

        [HasOne]
        public virtual User User { get; set; }
    }
}
